File: commutil/src/main/java/fz/gb/commutil/network/NetworkUtil.java
/*
 * Copyright (C) Globalegrow E-Commerce Co. , Ltd. 2007-2018.
 * All rights reserved.
 * This software is the confidential and proprietary information
 * of Globalegrow E-Commerce Co. , Ltd. ("Confidential Information").
 * You shall not disclose such Confidential Information and shall
 * use it only in accordance with the terms of the license agreement
 * you entered into with Globalegrow.
 */

package fz.gb.commutil.network;

import android.app.Activity;
import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.support.v4.app.Fragment;
import android.widget.Toast;

import fz.gb.commutil.R;
import fz.gb.commutil.log.CLog;
import fz.gb.commutil.toast.ToastUtil;


/**
 * 网络工具类
 *
 * @author dingpeihua
 * @version 1.0
 * @date 2017/11/8 10:18
 */
public final class NetworkUtil {
    private NetworkUtil() {
        throw new AssertionError();
    }


    /**
     * 检测网络是否连接
     *
     * @param context
     * @return 是返回true，否返回false
     */
    public static boolean isNetworkConnected(Context context) {
        if (context != null) {
            ConnectivityManager manager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
            NetworkInfo networkInfo = manager.getActiveNetworkInfo();
            boolean isConnected = networkInfo != null && networkInfo.isConnected();
            CLog.d("isConnected = " + isConnected);
            return isConnected;
        } else {
            throw new NullPointerException("activity is null");
        }
    }


    /**
     * 检测网络是否连接
     *
     * @param context
     * @param showNetworkErrorTips 是否提示显示网络错误信息，是表示显示，否表示不显示
     * @return 是返回true，否返回false
     */
    public static boolean isNetworkConnected(Context context, boolean showNetworkErrorTips) {
        ConnectivityManager manager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo networkInfo = manager.getActiveNetworkInfo();
        boolean isConnected = networkInfo != null && networkInfo.isConnected();
        if (!isConnected && showNetworkErrorTips) {
            showToast(context.getString(R.string.im_tips_check_network));
        }
        return isConnected;
    }

    /**
     * 检测网络是否连接
     *
     * @param fragment
     * @return 是返回true，否返回false
     */
    public static boolean isNetworkConnected(Fragment fragment) {
        if (fragment != null) {
            return isNetworkConnected(fragment.getActivity());
        } else {
            throw new NullPointerException("fragment is null");
        }
    }

    /**
     * 检测网络是否连接
     *
     * @param fragment
     * @return 是返回true，否返回false
     */
    public static boolean isNetworkConnected(Fragment fragment, boolean showNetworkErrorTips) {
        if (fragment != null && fragment.getActivity() != null) {
            return isNetworkConnected(fragment.getActivity(), showNetworkErrorTips);
        }
        return false;
    }


    /**
     * 判断是不是使用WiFi链接
     *
     * @author dingpeihua
     * @date 2016/10/19 16:37
     * @version 1.0
     */
    public static boolean isConnectedWifi(Context context) {
        return Connectivity.isConnectedWifi(context);
    }

    /**
     * 判断是不是使用手机移动网络链接
     *
     * @author dingpeihua
     * @date 2016/10/19 16:38
     * @version 1.0
     */
    public static boolean isConnectedMobile(Context context) {
        return Connectivity.isConnectedMobile(context);
    }

    /**
     * 检测网络是否连接
     *
     * @param showNetworkErrorTips true 在没有网络时提示，否则不提示
     * @author dingpeihua
     * @date 2016/10/19 16:42
     * @version 1.0
     */
    public static boolean isConnected(Context context, boolean showNetworkErrorTips) {
        boolean isConnect = Connectivity.isConnected(context);
        if (!isConnect && showNetworkErrorTips) {
            showToast(context.getString(R.string.im_tips_check_network));
        }
        return isConnect;
    }

    /**
     * 检测网络是否连接
     *
     * @author dingpeihua
     * @date 2016/10/19 16:42
     * @version 1.0
     */
    public static boolean isConnected(Fragment fragment) {
        if (fragment != null) {
            boolean isConnect = Connectivity.isConnected(fragment.getActivity());
            if (!isConnect) {
                showToast(fragment.getString(R.string.im_tips_check_network));
            }
            return isConnect;
        }
        return false;
    }

    /**
     * 检测网络是否连接
     *
     * @author dingpeihua
     * @date 2016/10/19 16:42
     * @version 1.0
     */
    public static boolean isConnected(Context context) {
        if (context != null) {
            boolean isConnect = Connectivity.isConnected(context.getApplicationContext());
            if (!isConnect) {
                showToast(context.getString(R.string.im_tips_check_network));
            }
            return isConnect;
        }
        return false;
    }

    public static boolean isConnectedFast(Context context) {
        return Connectivity.isConnectedFast(context);
    }

    public static @NetworkType.Type String  getNetworkType(Context context) {
        return Connectivity.getNetworkType(context);
    }

    public static @NetworkType.Type String  getNetworkType(NetworkInfo info) {
        return Connectivity.getNetworkType(info);
    }//Must specify a repository for deployment

    public static void showToast(final String message) {
        ToastUtil.showShortMessage(message);
    }
}

File: commutil/src/main/java/com/fz/common/network/NetworkUtil.kt
package com.fz.common.network

import android.content.Context
import androidx.fragment.app.Fragment
import com.fz.common.ContextInitializer
import com.fz.common.R
import com.fz.common.text.isNonEmpty
import com.fz.toast.ToastCompat

fun Context.isConnected(): Boolean {
    return Connectivity.isConnected(this)
}

fun Context.isConnected(showNetworkErrorTips: Boolean): Boolean {
    return NetworkUtil.isConnected(this, showNetworkErrorTips)
}

fun Fragment?.isConnected(showNetworkErrorTips: Boolean): Boolean {
    return NetworkUtil.isConnected(this, showNetworkErrorTips)
}

fun Fragment?.isConnected(): Boolean {
    return NetworkUtil.isConnected(this)
}

fun Context.getNetworkType(): NetworkType {
    return NetworkUtil.getNetworkType(this)
}

fun Context.isConnectedWifi(context: Context): Boolean {
    return NetworkUtil.isConnectedWifi(context)
}

fun Context.isConnectedMobile(context: Context): Boolean {
    return NetworkUtil.isConnectedMobile(context)
}

fun Context.isConnectionFast(context: Context): Boolean {
    return NetworkUtil.isConnectionFast(context)
}

/**
 * 网络工具类
 *
 * @author dingpeihua
 * @version 1.0
 * @date 2017/11/8 10:18
 */
object NetworkUtil {

    /**
     * 检测网络是否连接
     *
     * @author dingpeihua
     * @date 2016/10/19 16:42
     * @version 1.0
     */
    @JvmStatic
    fun isConnected(): Boolean {
        return ContextInitializer.mContext.let { Connectivity.isConnected(it) }
    }

    /**
     * 检测网络是否连接
     *
     * @author dingpeihua
     * @date 2016/10/19 16:42
     * @version 1.0
     */
    @JvmStatic
    fun isConnected(context: Context): Boolean {
        return Connectivity.isConnected(context)
    }

    /**
     * 检测网络是否连接
     *
     * @param showNetworkErrorTips true 在没有网络时提示，否则不提示
     * @author dingpeihua
     * @date 2016/10/19 16:42
     * @version 1.0
     */
    @JvmStatic
    fun isConnected(context: Context, showNetworkErrorTips: Boolean): Boolean {
        val isConnect = isConnected(context)
        if (!isConnect && showNetworkErrorTips) {
            showToast(context, context.getString(R.string.tips_check_network))
        }
        return isConnect
    }

    /**
     * 检测网络是否连接
     *
     * @author dingpeihua
     * @date 2016/10/19 16:42
     * @version 1.0
     */
    @JvmStatic
    fun isConnected(fragment: Fragment?, showNetworkErrorTips: Boolean): Boolean {
        if (fragment != null) {
            val isConnect = isConnected(fragment.requireContext())
            if (!isConnect && showNetworkErrorTips) {
                showToast(fragment.context, fragment.getString(R.string.tips_check_network))
            }
            return isConnect
        }
        return false
    }

    /**
     * 检测网络是否连接
     *
     * @author dingpeihua
     * @date 2016/10/19 16:42
     * @version 1.0
     */
    @JvmStatic
    fun isConnected(fragment: Fragment?): Boolean {
        return isConnected(fragment, false)
    }

    @JvmStatic
    fun getNetworkType(context: Context): NetworkType {
        return Connectivity.getNetworkType(context)
    }

    @JvmStatic
    fun isConnectedWifi(context: Context): Boolean {
        return Connectivity.isConnectedWifi(context)
    }

    @JvmStatic
    fun isConnectedMobile(context: Context): Boolean {
        return Connectivity.isConnectedMobile(context)
    }

    @JvmStatic
    fun isConnectionFast(context: Context): Boolean {
        return Connectivity.isConnectionFast(context)
    }

    private fun showToast(context: Context?, message: String?) {
        if (message.isNonEmpty()) {
            ToastCompat.showMessage(context, message)
        }
    }
}